### PR TITLE
devices: add index to image_id and optimise the negative query.

### PR DESCRIPTION
**Pull request by ldjebran**
we are affected to a performance issue on devices view

Fixes THEEDGE-2915

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/routes/devices.go [hunk 1]
```diff
@@ -313,7 +313,7 @@ func GetDeviceDBInfo(w http.ResponseWriter, r *http.Request) {
 // GetDevicesView returns all data needed to display customers devices
 func GetDevicesView(w http.ResponseWriter, r *http.Request) {
 	contextServices := dependencies.ServicesFromContext(r.Context())
-	tx := devicesFilters(r, db.DB.Debug()).Where("image_id IS NOT NULL AND image_id != 0")
+	tx := devicesFilters(r, db.DB.Debug()).Where("image_id > 0")
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/routes/devices.go]: done

**Review by de1987 — APPROVED** [review]

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Comment by thom-at-redhat** [comment]
@mergifyio refresh

**Comment by thom-at-redhat** [comment]
@mergifyio unqueue

**Comment by thom-at-redhat** [comment]
@mergifyio requeue edge-api

**Comment by thom-at-redhat** [comment]
@mergifyio unqueue

**Pull request merged**